src/Cardano/Types/Transaction.purs

```diff
@@ -0,0 +1,799 @@
+
+newtype Vkeywitness = Vkeywitness (Vkey /\ Ed25519Signature)
+
+derive instance Generic Vkeywitness _
+derive newtype instance Eq Vkeywitness
+
+instance Show Vkeywitness where
+  show = genericShow
+
+newtype Vkey = Vkey PublicKey
+
+derive instance Generic Vkey _
+derive instance Newtype Vkey _
+derive newtype instance Eq Vkey
+derive newtype instance Ord Vkey
+
+instance Show Vkey where
+  show = genericShow
+
+newtype PublicKey = PublicKey Bech32String
+
+derive instance Generic PublicKey _
+derive instance Newtype PublicKey _
+derive newtype instance Eq PublicKey
+derive newtype instance Ord PublicKey
+
+instance Show PublicKey where
+  show = genericShow
+
+newtype Ed25519Signature = Ed25519Signature Bech32String
+
+derive instance Generic Ed25519Signature _
+derive newtype instance Eq Ed25519Signature
+derive newtype instance Ord Ed25519Signature
+
+instance Show Ed25519Signature where
+  show = genericShow
+
+newtype Redeemer = Redeemer
+  { tag :: RedeemerTag
+  , index :: BigInt
+  , data :: PlutusData
+  , exUnits :: ExUnits
+  }
+
+derive instance Generic Redeemer _
+derive newtype instance Eq Redeemer
+derive newtype instance Ord Redeemer
+
+instance Show Redeemer where
+  show = genericShow
+
+newtype AuxiliaryData = AuxiliaryData
+  { metadata :: Maybe GeneralTransactionMetadata
+  , nativeScripts :: Maybe (Array NativeScript)
+  , plutusScripts :: Maybe (Array PlutusScript)
+  }
+
+derive newtype instance Eq AuxiliaryData
+derive instance Generic AuxiliaryData _
+
+instance Show AuxiliaryData where
+  show = genericShow
+
+instance Semigroup AuxiliaryData where
+  append (AuxiliaryData ad) (AuxiliaryData ad') =
+    AuxiliaryData
+      { metadata: ad.metadata <> ad'.metadata
+      , nativeScripts: lift2 union ad.nativeScripts ad'.nativeScripts
+      , plutusScripts: lift2 union ad.plutusScripts ad'.plutusScripts
+      }
+
+instance Monoid AuxiliaryData where
+  mempty = AuxiliaryData
+    { metadata: Nothing
+    , nativeScripts: Nothing
+    , plutusScripts: Nothing
+    }
+
+data NativeScript
+  = ScriptPubkey Ed25519KeyHash
+  | ScriptAll (Array NativeScript)
+  | ScriptAny (Array NativeScript)
+  | ScriptNOfK Int (Array NativeScript)
+  | TimelockStart Slot
+  | TimelockExpiry Slot
+
+derive instance Eq NativeScript
+derive instance Generic NativeScript _
+
+instance Show NativeScript where
+  show x = genericShow x
+
+newtype TransactionOutput = TransactionOutput
+  { address :: Address
+  , amount :: Value
+  , dataHash :: Maybe DataHash
+  }
+
+derive instance Generic TransactionOutput _
+derive instance Newtype TransactionOutput _
+derive newtype instance Eq TransactionOutput
+
+instance Show TransactionOutput where
+  show = genericShow
+
+newtype UtxoM = UtxoM Utxo
+
+derive instance Generic UtxoM _
+derive instance Newtype UtxoM _
+derive newtype instance Eq UtxoM
+
+instance Show UtxoM where
+  show = genericShow
+
+type Utxo = Map TransactionInput TransactionOutput
+
+-- Option<Certificates>,
```

Comment:
It's really time for these comments to go haha. They've somehow survived since the beginning